crates/server/src/admin/handlers/challenge.rs

```diff
@@ -60,15 +61,15 @@ pub async fn request_challenge_handler(
 }
 
 fn generate_challenge(
-    application_id: Option<String>,
+    context_id: Option<ContextId>,
     keypair: &Keypair,
 ) -> Result<NodeChallenge, ApiError> {
     let random_bytes = generate_random_bytes();
     let encoded = STANDARD.encode(&random_bytes);
 
     let node_challenge_message = NodeChallengeMessage {
         nonce: encoded,
-        application_id: application_id.unwrap_or_default(),
+        context_id,
```

Comment:
Why was this `application_id.unwrap_or_default()` before? Should it also be `context_id.unwrap_or_default()` because of Option or not?

Reply:
Connected to https://github.com/calimero-network/core/pull/410#discussion_r1652280855

Figured it's best to retain the option than to assume a default

Just need to ensure the associated behaviours on admin UI & apps are consistent with this change